internal/v1/handler_blueprints.go

```diff
@@ -300,6 +300,23 @@ func (h *Handlers) UpdateBlueprint(ctx echo.Context, blueprintId uuid.UUID) erro
 		})
 	}
 
+	if blueprintRequest.Customizations.Users != nil {
+		for _, user := range *blueprintRequest.Customizations.Users {
+			// Make sure every user has either ssh key or password set
+			if user.Password == nil && user.SshKey == nil {
+				return ctx.JSON(http.StatusUnprocessableEntity, HTTPErrorList{
+					Errors: []HTTPError{{
+						Title:  "Invalid user",
+						Detail: "User must have either a password or an SSH key set.",
+					}},
+				})
+			}
+			// TODO: User password "<REDACTED>"
+			// 1. Previous password exists = set old password
```

Comment:
I guess we'll need to do this before marshaling the body.
Fetching the "old" Blueprint from DB should [be easy by db.GetBlueprint](https://github.com/osbuild/image-builder/blob/e8780a2796b470aa5c5f9e8b44c2c68af60dd52b/internal/db/db.go#L79).

Then just a bit tedious for going through both the structures and comparing user by user, but I hope it shouldn't be that bad :thinking: